source/agora/consensus/protocol/Nominator.d

```diff
@@ -483,8 +483,9 @@ extern(D):
         }
         if (!verify(public_key, envelope.signature.toSignature(), challenge))
         {
-            // If it fails signature verification, it might not originate from said key
-            log.trace("Envelope failed signature verification for {}", public_key);
+            log.trace("INVALID Envelope signature {} for public key {} " ~
+                "challenge {}",
+                envelope.signature, public_key, challenge.toString(PrintMode.Clear));
```

Comment:
?

Reply:
It may be an envelope from a bad node so we log at trace level the details. The challenge is a Scalar so we need to print it in the clear.

Reply:
I mean, we shortened this log message just last PR, and you are essentially reverting those changes here

Reply:
Reverted.